neuraxle/base.py

```diff
@@ -3445,9 +3456,25 @@ class ForceHandleMixin:
     """
 
     def __init__(self, cache_folder=None):
+        MixinForBaseTransformer.__init__(self)
         if cache_folder is None:
             cache_folder = DEFAULT_CACHE_FOLDER
         self.cache_folder = cache_folder
+        if isinstance(self, _FittableStep):
+            self._ensure_method_overriden("_fit_data_container", _FittableStep)
+            self._ensure_method_overriden("_fit_transform_data_container", _FittableStep)
+        self._ensure_method_overriden("_transform_data_container", _TransformerStep)
+
+    def _ensure_method_overriden(self, method_name, original_cls):
+        """
+        Asserts that a given method of current instance overrides the default one defined in a given class. We assume that current instance inherits from the given class but do not test it (MixinForBaseTransformer test inheritance to BaseTransfromer already).
+
+        :param method_name:
+        :param original_cls:
+        :return:
+        """
+        if original_cls.__dict__[method_name] == getattr(self, method_name).__func__:
+            raise NotImplementedError(f"Please define method {method_name} in {self.__class__.__name__}.")
```

Comment:
Thinking twice despite our pair programming review session together, an even clearer error would be possible. Such as explaining the cause of this error itself and the link with the present class' role and purpose.

Reply:
What do you think of the following : 
"The ForceHandleMixin class overrides fit, transform and fit_transform to force a call on their handler method counterparts. Failure to redefine basic _fit_data_container, _transform_data_container and _fit_transform_data_container methods may cause an infinite loop. Please define  {method_name} in {self.__class__.__name__}."

Reply:
Your suggestion sounds good ! 

Note that the " ` " character (and especially its triple-repeated counterpart) can be used on GitHub, StackOverflow, fb Messenger, and even Telegram. It is a general markdown code block escape `like this`. But you probably know ;)